pkg/tls/cred_test.go

```diff
@@ -47,3 +48,27 @@ func TestCredEncodeCertificateAndTrustChain(t *testing.T) {
 		t.Errorf("Encoded Certificate And TrustChain does not match expected output")
 	}
 }
+
+func TestCrtExpiry(t *testing.T) {
+	root := newRoot(t)
+	rootTrust := root.Cred.Crt.CertPool()
+
+	cred, err := root.GenerateEndEntityCred("expired.test")
+	if err != nil {
+		t.Fatalf("failed to create end entity cred: %s", err)
+	}
+
+	crt, err := DecodePEMCrt(cred.Crt.EncodePEM())
+	if err != nil {
+		t.Fatalf("Failed to decode PEM Crt: %s", err)
+	}
+
+	//explicitly kill the certificate
+	crt.Certificate.NotBefore = time.Now().Add(time.Duration(100000000000))
+
+	if err := crt.Verify(rootTrust, "expired.test"); err != nil {
+		t.Logf("could not process CSR because of CA cert validation failure: certificate: %s", err)
+	} else {
+		t.Fatalf("failed to detect expired certificates")
+	}
+}
```

Comment:
This test is a bit strange. In your code you are emitting an event that includes the now time and the expiration time. But here you are killing the cert by moving the NotBefore date further away in time, effectivelly making it not expired but simply not valid yet.. Also you are simply testing that an error is present. I think you want to test for the specific type of error. So the following cases make sense to be covered: 
1. Test that the cert is not valid yet - NotBefore now + 10 days, notAfter now + 20 days 
2. Test that the cert is expired - NotBefore now - 20 daysm NotAfter now - 10 days
3. Make sure you get an exact error, not simply any error.

Reply:
@zaharidichev Sure, I understood. I just wanted to clear my understanding a bit before I write the tests.
I looked into the `x509` library to understand what triggers the Expiry error:
```
	if now.Before(c.NotBefore) || now.After(c.NotAfter) {
		return CertificateInvalidError{c, Expired, ""}
	}
```
The Expiry error is triggered when 1) The current time is before `NotBefore` (the certificate is not valid yet), or 2) the current time is after `NotAfter` (the current time has expired).
Either of the cases will cause the certificate to be Expired. I seem to have written only the case (1). Need to cover (2) as well, right?

Reply:
@zaharidichev Made the changes. Could you have a look and let me know if I've got the tests right?